rust/projects/file-io/src/kv.rs

```diff
@@ -40,4 +66,28 @@ impl KvStore {
     pub fn get(&self, key: String) -> Result<Option<String>> {
         Ok(self.map.get(&key).cloned())
     }
+
+    fn load_from_wal(wal_path: impl AsRef<Path>) -> Result<HashMap<String, String>> {
+        let mut map = HashMap::new();
+        let reader = BufReader::new(File::open(wal_path)?);
+        let stream = Deserializer::from_reader(reader).into_iter::<SetCommand>();
+        for set_cmd in stream {
+            let set_cmd = set_cmd?;
+            map.insert(set_cmd.key, set_cmd.value);
+        }
```

Comment:
I'm glad this code was so easy. I was afraid serde wouldn't deserialize multiple records in sequence like that.